src/wallet/wallettool.cpp

```diff
@@ -112,7 +112,7 @@ static bool SalvageWallet(const fs::path& path)
     // Initialize the environment before recovery
     bilingual_str error_string;
     try {
-        WalletBatch::VerifyEnvironment(path, error_string);
+        database->Verify(error_string);
```

Comment:
In commit "walletdb: Combine VerifyDatabaseFile and VerifyEnvironment" (ae1a9d9ebbe1b53fd245fddd1907ca90fe3a7abe)

It doesn't seem like there is a point to calling Verify here in `SalvageWallet` when `ExecuteWalletToolFunc` already called it immediately before calling this. It seems like this `SalvageWallet` function could just be dropped and `ExecuteWalletToolFunc` just call `RecoverDatabaseFile` directly.

Reply:
Since the database is closed before we reach this function, we need to reopen it before doing the recovery function. We could remove this and only close the database for `info` but that seems a bit like a hack.

Maybe something to think about for a followup.

Reply:
In commit "walletdb: Combine VerifyDatabaseFile and VerifyEnvironment" (8f1bcf8b7b6e47c05f2e43dd98ec3505b888d8b3)

> Since the database is closed before we reach this function, we need to reopen it before doing the recovery function. We could remove this and only close the database for `info` but that seems a bit like a hack.
> 
> Maybe something to think about for a followup.

It seems like the unintended change in behavior described https://github.com/bitcoin/bitcoin/pull/19324#discussion_r442532065 is back.

Instead of salvage command just recovering wallet data, now it tries to verify before recovering. Maybe this change is good because verifying before recovering is good for some reason. Maybe it is bad because it prevents salvage command from working some cases. Or maybe it is ok but slows the salvage operation down unnecessarily. I can't tell if it's good or bad but it seems unintended and not mentioned in the PR description or any commit message and ignored in reviews.

It looks like previous behavior could be restored by deleting the `SalvageWallet` function and just calling `RecoverDatabaseFile` function instead (previously suggested https://github.com/bitcoin/bitcoin/pull/19324#discussion_r442527386)

Reply:
Hmm. Seems like there was a rebase error at some point. I'll make a followup that handles this and a few other related suggestions.